luskan/src/main/java/org/opencadc/luskan/QueryRunnerImpl.java

```diff
@@ -103,4 +114,48 @@ protected DataSource getQueryDataSource() throws Exception {
         log.debug("Data Source name: jdbc/tapuser");
         return DBUtil.findJNDIDataSource("jdbc/tapuser");
     }
+
+    /**
+     *
+     */
+    @Override
+    public void run() {
```

Comment:
This is not the most upstream place to block access. In TAP, all jobs are uws jobs (sync and async) so the right place to check for global access is in JobPersistence.put(Job) - check auth before job creation means you get "permission denied" on the original job create and don't put a job in the database, rather than only getting it when running a job (where, if you are lucky this throw will give the right error from an async job but I doubt it).

Reply:
Added an AuthJobPersistence that extends PostgresJobPersistence and does the auth check before putting the job.